### PR TITLE
Remove reactor remote cluster state changes

### PR DESCRIPTION
**Pull request by gawarz**
This PR remove reactor from remote cluster state

### CODE REVIEW

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/RemoteServices.kt [hunk 1]
```diff
@@ -23,34 +23,38 @@ class RemoteServices(
...
-            .doOnCancel {
-                meterRegistry.counter("cross-dc-synchronization.cancelled").increment()
-                logger.warn("Cancelling cross dc sync")
+        return Flux.create({ sink ->
+            scheduler.scheduleWithFixedDelay({
+                meterRegistry.timer("sync-dc.get-multi-cluster-states.time").record {
+                    getChanges(sink::next, interval)
+                }
+            }, 0, interval, TimeUnit.SECONDS)
+        }, FluxSink.OverflowStrategy.LATEST)
```
Comment (gawarz) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/RemoteServices.kt:35]: add doOnCancel - metric

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/RemoteServices.kt [hunk 2]
```diff
@@ -23,34 +23,38 @@ class RemoteServices(
...
+    }
+
+    private fun getChanges(sink: (MultiClusterState) -> Unit, interval: Long) {
+        remoteClusters
+            .map { cluster -> clusterWithControlPlaneInstances(cluster) }
+            .filter { (_, instances) -> instances.isNotEmpty() }
+            .map { (cluster, instances) -> getClusterState(instances, cluster) }
+            .mapNotNull { it.get(interval, TimeUnit.SECONDS) }
```
Comment (gawarz) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/RemoteServices.kt:43]: how long we want to wait?

File: envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/RemoteServices.kt [hunk 3]
```diff
@@ -23,34 +23,38 @@ class RemoteServices(
...
+        return Flux.create({ sink ->
+            scheduler.scheduleWithFixedDelay({
+                meterRegistry.timer("sync-dc.get-multi-cluster-states.time").record {
+                    getChanges(sink::next, interval)
+                }
+            }, 0, interval, TimeUnit.SECONDS)
+        }, FluxSink.OverflowStrategy.LATEST)
+    }
+
+    private fun getChanges(sink: (MultiClusterState) -> Unit, interval: Long) {
```
Comment (gawarz) [inline envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/synchronization/RemoteServices.kt:38]: rename to consume

**Review by Automaat — APPROVED** [review]

**Comment by Automaat** [comment]
Looks great, can we test it on some real envionment before merging? To check how it affects performance of our control plane?

**Review by Ferdudas97 — APPROVED** [review]

**Review by Automaat — DISMISSED** [review]

**Review by Automaat — APPROVED** [review]

**Review by Ferdudas97 — APPROVED** [review]

**Pull request merged**